### PR TITLE
feat(core): Create a react modal wizard

### PR DESCRIPTION
**Pull request by jrsquared**
A react version of the v2modalWizard.component

TODO: Form validation is still pretty rudimentary; add support for `yup` or something more model based.

### CODE REVIEW

File: app/scripts/modules/core/src/modal/wizard/WizardModal.tsx [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+      dirtyPages: new Set<string>(),
+      errorPages: new Set<string>(),
+      formInvalid: false,
+      pages: [],
+    };
+
+  }
+
+  private setCurrentPage(pageState: IWizardPageData): void {
+    // pageState.element.scrollIntoView({ behavior: 'smooth' });
```
[the remark this answers is not in the document]
Comment (jrsquared) [reply app/scripts/modules/core/src/modal/wizard/WizardModal.tsx]: scrollIntoView is inconsistent in placement and is not a great experience.

File: app/scripts/modules/core/src/presentation/main.less [hunk 2]
```diff
@@ -1165,6 +1180,12 @@ select.input-sm {
   padding-left: 14px;
 }
 
+.Select.is-open { z-index: 99999; }
+
+.sortable-helper {
+  z-index: 99999;
```
Comment (archana-s) [inline app/scripts/modules/core/src/presentation/main.less]: same comment as above.

File: app/scripts/modules/core/src/modal/wizard/WizardModal.tsx [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+    const dirtyPages = new Set(this.state.dirtyPages);
+    if (dirty) {
+      dirtyPages.add(name);
+    } else {
+      dirtyPages.delete(name);
+    }
+    this.setState({ dirtyPages });
+  }
+
+  public componentDidMount(): void {
```
[the remark this answers is not in the document]
Comment (jrsquared) [reply app/scripts/modules/core/src/modal/wizard/WizardModal.tsx]: We haven't adopted a formal standard, but we probably should!

File: app/scripts/modules/core/src/modal/wizard/WizardModal.tsx [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+    default: !pageState.props.done,
+    dirty,
+    current,
+    done: pageState.props.done,
+  });
+  const handleClick = () => { onClick(pageState); };
+
+  return (
+    <li className={className}>
+      <a className="clickable" onClick={handleClick}>{pageState.label}</a>
```
[the remark this answers is not in the document]
Comment (jrsquared) [reply app/scripts/modules/core/src/modal/wizard/WizardModal.tsx]: done

File: app/scripts/modules/core/src/modal/buttons/ModalClose.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+export interface IModalCloseProps {
+  dismiss: () => void;
+}
+
+export class ModalClose extends React.Component<IModalCloseProps> {
+  public render() {
+    return (
+      <div className="modal-close close-button pull-right">
+        <a className="btn btn-link clickable" onClick={this.props.dismiss}>
```
[the remark this answers is not in the document]
Comment (jrsquared) [reply app/scripts/modules/core/src/modal/buttons/ModalClose.tsx]: done

File: app/scripts/modules/core/src/modal/wizard/WizardModal.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (jrsquared) [reply app/scripts/modules/core/src/modal/wizard/WizardModal.tsx]: There are a lot of custom rules for this specific case shared with the angular modal... I'm going to keep it an <a> for now, and keep in mind we should switch to a button once we finish conversion.

File: app/scripts/modules/core/src/modal/wizard/WizardModal.tsx [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+                      {this.getFilteredChildren().map((child: React.ReactElement<any>) => {
+                        return React.cloneElement(child, { ...props, dirtyCallback: this.dirtyCallback, onMount: this.onMount, revalidate: () => this.revalidate(props.values, props.setErrors) });
+                      })}
+                    </div>
+                  </div>
+                </div>
+              </Modal.Body>
+              <Modal.Footer>
+                <button
+                  className="btn btn-default"
```
[the remark this answers is not in the document]
Comment (jrsquared) [reply app/scripts/modules/core/src/modal/wizard/WizardModal.tsx]: Talked offline, will hold off until styleguide has been updated to account for this usecase.

**Pull request merged**